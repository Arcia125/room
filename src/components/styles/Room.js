import styled from 'styled-components';

const StyledRoom = styled.div`
  /* This line is giving padding to both chatbox and room-page... */
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-template-areas:
    'search-bar room-name'
    'list' 'messages';
  .chatbox {
    &__user-list {
      grid-area: 'list';
      list-style: none;
      width: 45rem;
      border: 1px solid gray;
      border-radius: 2px;
    }
    &__input {
      &--field {
        height: 4.8rem;
        /* padding: .8rem; */
      }
      &--button {
      }
    }
    &__header {
      color: ${p => p.theme.colors.grayDark};
      font-family: 'Poppins', sans-serif;
      font-weight: 500;
      font-size: 3.2rem;
    }

    &__message-list {
      grid-area: message-list;
      border: 1px solid gray;
      width: 90rem;
      min-height: 70rem;
      list-style: none;
      &--item {
      }
    }
  }

  .room-page {
    padding: 4rem;
    &__header {
      color: ${p => p.theme.colors.grayDark};
      font-family: 'Poppins', sans-serif;
      font-weight: 500;
      font-size: 4.8rem;
    }
  }
`;

export default StyledRoom;
