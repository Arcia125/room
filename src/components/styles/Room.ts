import styled from 'styled-components';

const StyledRoom = styled.div`
  /* This line is giving padding to both chatbox and room-page... */
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 3fr;
  grid-template-areas:
    'search-bar room-name'
    'list' 'messages';
  background-color: ${'#EFEEF8'};
  .chatbox {
    max-width: 90rem;
    &__user-list {
      grid-area: 'list';
      list-style: none;
      width: 45rem;
      border: 1px solid gray;
      border-radius: 2px;
    }

    &__header {
      color: ${'#454D57'};
      font-family: 'Montserrat', sans-serif;
      padding-bottom: 1.6rem;
      margin-bottom: 1.6rem;
      border-bottom: 2px solid #e4e6ee;
      font-weight: 500;
      font-size: 2rem;
    }

    &__message-list {
      grid-area: message-list;
      width: 90rem;
      min-height: 70rem;
      list-style: none;
      display: grid;
      grid-gap: 2rem;
      grid-template-rows: repeat(9, min-content);
      &--item {
        font-family: 'Montserrat', sans-serif;
        font-weight: 500;
        font-size: 1.6rem;
        background-color: ${'#EAE8ED'};
        padding: 1.6rem;
        border-radius: 18px;
      }

      /* styles for messages from user */

      /* styles for messages from others */
    }
    &__input {
      height: 5.4rem;
      border-radius: 0 0 4px 4px;
      padding: 0 16px;
      background-color: #fff;
      display: flex;
      justify-content: space-between;
      align-items: center;
      &--field {
        border: none;
        width: 100%;
        height: 100%;
        font-size: 1.6rem;
        &::placeholder {
          font-family: 'Montserrat', sans-serif;
          font-weight: 600;
          font-size: 1.6rem;
          color: #c6c8cb;
        }
        /* padding: .8rem; */
      }
      &--button {
        background: radial-gradient(
          farthest-corner at 0px 5px,
          #2d8df2,
          #9198e5
        );
        transform: rotate(-45deg);
        width: 4rem;
        height: 4rem;
        border: none;
        border-radius: 50%;
        cursor: pointer;
        .material-icons {
          transform: translate(2px);
          font-size: 26px;
          color: #fff;
        }
      }
    }
  }

  .room-page {
    padding: 4rem;
    &__header {
      color: ${'#1f1f1f'};
      font-family: 'Poppins', sans-serif;
      font-weight: 500;
      font-size: 4.8rem;
    }
  }
`;

export default StyledRoom;
