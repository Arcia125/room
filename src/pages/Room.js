import React from 'react';
import PropTypes from 'prop-types';
import { useQuery, useMutation } from '@apollo/react-hooks';

import { GET_ROOM } from '../graphql/getRoom';
import { SEND_MESSAGE } from '../graphql/sendMessage';

const Room = ({ match }) => {
  const [userMessage, setUserMessage] = React.useState('');
  const handleInputChange = event => setUserMessage(event.target.value);

  const roomId = match.params.roomId;

  const roomQuery = useQuery(GET_ROOM, { variables: { id: roomId } });

  const room = roomQuery.data && roomQuery.data.room;

  const handleSendMessageComplete = () => setUserMessage('');

  const [sendMessage, sendMessageMutation] = useMutation(SEND_MESSAGE, {
    variables: { roomId, content: userMessage },
    onCompleted: handleSendMessageComplete
  });

  const handleSend = event => {
    if (userMessage.length) {
      sendMessage();
    }
  };

  let content;
  if (roomQuery.loading) {
    content = 'loading...';
  }

  if (roomQuery.error) {
    content = roomQuery.error.message;
  }

  if (room) {
    content = (
      <main className="room-page-content">
        <h1 className="room-page-content__header">{room.name}</h1>
        <ul className="message-list">
          {room.messages.map(message => (
            <li className="message-list__item">{message.content}</li>
          ))}
        </ul>
        <input
          className="room-page-content__user-input"
          value={userMessage}
          onChange={handleInputChange}
        />
        <button className="room-page-content__send-button" onClick={handleSend}>
          send
        </button>
      </main>
    );
  }

  return (
    <div className="room-page">
      <header className="room-page-header"></header>
      {content}
    </div>
  );
};

Room.propTypes = {};

export default Room;
