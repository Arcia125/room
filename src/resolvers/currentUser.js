import { getUser, setUser, getToken } from '../utils/user';
// let user = {
//   id: null,
//   username: null,
//   __typename: 'User'
// };

// let user = getUser();

const currentUser = {
  defaults: {
    currentUser: getUser()
  },
  resolvers: {
    Mutation: {
      saveUser: (_, { username, id, email, token }, { cache }) => {
        const user = { id, username, email, __typename: 'User' };
        console.log('saveUser resolver', user, token);
        // console.log('saving user ', user);
        // cache.writeQuery({
        //   query: GET_CURRENT_USER,
        //   data: { currentUser: user },
        // });

        setUser({ user, cache, token });

        return { user, token, __typename: 'AuthPayload' };
        // cache.writeData({
        //   currentUser: {
        //     id,
        //     username,
        //     __typename: 'User',
        //   },
        // });
      }
    },
    Query: {
      currentUser: (root, args, { cache }, info) => {
        // debugger;
        console.log('getting current user');
        // const queryResult = cache.readQuery({
        //   query: GET_CURRENT_USER,
        // });
        // console.log('readQuery', queryResult);
        const user = getUser();
        return user ? user : null;
        // return {
        //   user: getUser(),
        //   token: getToken(),
        // };
      }
    }
  }
};

export { currentUser };
