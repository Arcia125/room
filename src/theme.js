const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px'
  }
};

export { theme };
