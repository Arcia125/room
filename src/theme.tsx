import React, { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  spacing: {
    0: '0px',
    1: '8px',
    2: '16px',
    3: '32px',
  },
  colors: {
    gray: '#B3B2C0',
    grayDark: '#1f1f1f',
    lightBlue: '#AEAFE8',
    blue: '#AC2FC9',
    lightPurple: '#EFEEF8',
    purple: '#9056D4',
    pink: '#DD6589',
    bleh: '#8A4966',
  },
};

const Provider: React.FunctionComponent<{ children: ReactElement }> = ({
  children,
  ...restProps
}) => (
  <ThemeProvider theme={theme} {...restProps}>
    {children}
  </ThemeProvider>
);

export { theme, Provider };
