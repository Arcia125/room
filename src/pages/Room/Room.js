import React from 'react';
import PropTypes from 'prop-types';
import { useQuery, useMutation, useSubscription } from '@apollo/react-hooks';

import StyledRoom from '../../components/styles/Room';
import { GET_ROOM } from '../../graphql/getRoom';
import { SEND_MESSAGE } from '../../graphql/sendMessage';
import { Button } from '../../components/styles/Button';
import { NEW_ROOM_MESSAGE } from '../../graphql/newRoomMessage';

const useRoom = roomId => useQuery(GET_ROOM, { variables: { id: roomId } });

const useActiveRoom = roomId => {
  const roomQuery = useRoom(roomId);

  const updateQuery = ({ subscriptionData, client }) => {
    if (!subscriptionData.data) return;

    const { newRoomMessage } = subscriptionData.data;

    // copy previous query data
    const newData = {
      ...roomQuery.data
    };

    // add new message to data
    newData.room.messages = [...newData.room.messages, newRoomMessage];

    // update query in cache
    const newQuery = {
      query: GET_ROOM,
      variables: roomQuery.variables,
      data: newData
    };
    client.writeQuery(newQuery);
  };

  const newMessageSubscription = useSubscription(NEW_ROOM_MESSAGE, {
    variables: { roomId },
    onSubscriptionData: updateQuery
  });

  return { roomQuery, newMessageSubscription };
};

const Room = ({
  match: {
    params: { roomId }
  }
}) => {
  const [userMessage, setUserMessage] = React.useState('');
  const handleInputChange = event => setUserMessage(event.target.value);

  // const roomId = match.params.roomId;

  const { roomQuery, newMessageSubscription } = useActiveRoom(roomId);

  const room = roomQuery.data && roomQuery.data.room;

  const handleSendMessageComplete = () => setUserMessage('');

  const [sendMessage, sendMessageMutation] = useMutation(SEND_MESSAGE, {
    variables: { roomId, content: userMessage },
    onCompleted: handleSendMessageComplete
  });

  const handleSend = () => {
    if (userMessage.length) {
      sendMessage();
    }
  };

  const submitIfEnter = event => {
    const enterPressed = event.keyCode === 13;
    if (!enterPressed) return;
    handleSend();
  };

  let content;
  if (roomQuery.loading) {
    content = 'loading...';
  }

  if (roomQuery.error) {
    content = roomQuery.error.message;
  }

  if (room) {
    content = (
      <StyledRoom className="chatbox">
        <h1 className="chatbox__header">{room.name}</h1>
        <div className="chatbox__user-list">
          <li>Kevin</li>
          <li>Kaitlyn</li>
          <li>Chelsie</li>
          <li>Donte</li>
        </div>
        <ul className="chatbox__message-list">
          {room.messages.map(message => (
            <li key={message.id} className="chatbox__message-list--item">
              {message.content}
            </li>
          ))}
        </ul>
        <div className="chatbox__input">
          <input
            className="room-page-content__user-input"
            value={userMessage}
            onChange={handleInputChange}
            onKeyDown={submitIfEnter}
          />
          <Button color="primary" onClick={handleSend}>
            {sendMessageMutation.loading ? 'sending...' : 'send'}
          </Button>
        </div>
      </StyledRoom>
    );
  }

  return (
    <StyledRoom className="room__page">
      <header className="room-page__header">Chat</header>
      {content}
    </StyledRoom>
  );
};

Room.propTypes = {
  match: PropTypes.object
};

export default Room;
