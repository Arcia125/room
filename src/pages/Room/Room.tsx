import React from 'react';
import { useMutation } from '@apollo/react-hooks';
import { RouteComponentProps } from 'react-router';

import StyledRoom from '../../components/styles/Room';
import UserList from '../../components/UserList';
import { SEND_MESSAGE } from '../../graphql/sendMessage';
import { Button } from '../../components/styles/Button';
import { useActiveRoom } from '../../hooks/useActiveRoom';

interface RoomMessage {
  id: string;
  content: string;
}

const Room: React.FunctionComponent<
  RouteComponentProps<{ roomId: string }>
> = ({
  match: {
    params: { roomId },
  },
}) => {
  const [userMessage, setUserMessage] = React.useState('');
  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    setUserMessage(event.target.value);

  const { roomQuery } = useActiveRoom(roomId);

  const room = roomQuery.data && roomQuery.data.room;

  const handleSendMessageComplete = () => setUserMessage('');

  const [sendMessage, sendMessageMutation] = useMutation(SEND_MESSAGE, {
    variables: { roomId, content: userMessage },
    onCompleted: handleSendMessageComplete,
  });

  const handleSend = () => {
    if (userMessage.length) {
      sendMessage();
    }
  };

  const submitIfEnter = (event: React.KeyboardEvent<HTMLInputElement>) => {
    const enterPressed = event.keyCode === 13;
    if (!enterPressed) return;
    handleSend();
  };

  let content;
  if (roomQuery.loading) {
    content = 'loading...';
  }

  if (roomQuery.error) {
    content = roomQuery.error.message;
  }

  if (room) {
    content = (
      <StyledRoom>
        <UserList users={room.users} />
        <div className="chatbox">
          <h1 className="chatbox__header">{room.name}</h1>
          <ul className="chatbox__message-list">
            {room.messages.map((message: RoomMessage) => (
              <li key={message.id} className="chatbox__message-list--item">
                {message.content}
              </li>
            ))}
          </ul>
          <div className="chatbox__input">
            <input
              className="chatbox__input--field"
              value={userMessage}
              onChange={handleInputChange}
              onKeyDown={submitIfEnter}
            />
            <Button
              className="chatbox__input--button"
              color="primary"
              onClick={handleSend}
            >
              {sendMessageMutation.loading ? 'sending...' : 'send'}
            </Button>
          </div>
        </div>
      </StyledRoom>
    );
  }

  return (
    <div className="room-page">
      <header className="room-page__header">Chat</header>
      {content}
    </div>
  );
};

export default Room;
