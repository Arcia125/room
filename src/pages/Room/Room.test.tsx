import React from 'react';
import { MockedProvider } from '@apollo/react-testing';
import { render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Room from '.';
import { Provider } from '../../theme';
import {
  getRoomMock,
  newRoomMessageMock,
  mockRoom,
  newRoomUserMock,
  joinRoomMock
} from '../../../shared/queryMocks/room';
import { mockRouteComponentProps } from '../../../shared/mocks/mockRouteComponentProps';
import { currentUserMockResolvers } from '../../../shared/queryMocks/currentUser';

const mocks = [getRoomMock, newRoomMessageMock, newRoomUserMock, joinRoomMock];

afterEach(cleanup);

describe('Room', () => {
  it('renders without crashing', async () => {
    const { container } = render(
      <MockedProvider
        addTypename
        mocks={mocks}
        resolvers={currentUserMockResolvers}
      >
        <Provider>
          <Room
            {...mockRouteComponentProps}
            match={{
              ...mockRouteComponentProps.match,
              params: { roomId: mockRoom.id }
            }}
          />
        </Provider>
      </MockedProvider>
    );
    expect(container).toBeInTheDocument();
  });
});
