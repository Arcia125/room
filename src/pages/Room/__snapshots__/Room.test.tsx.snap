// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room finishes loading initial queries 1`] = `
<DocumentFragment>
  <div
    class="sc-ifAKCX kViNAh"
  >
    <nav
      class="sc-EHOje hpFLPR"
    >
      <div
        class="user"
      >
        <img
          alt="avatar"
          src="https://source.unsplash.com/cqraK2a3Or8/50x50"
          style="border-radius: 50%;"
        />
        <p>
          Kevin
        </p>
      </div>
      <ul>
        <li>
          <button
            class="sc-bxivhb UGMhj"
            data-testid="create-room-button"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 20V14H14V20H19V12H22L12 3L2 12H5V20H10Z"
                fill="#AEB3C8"
              />
            </svg>
            create room
          </button>
        </li>
        <li>
          <button
            class="sc-bxivhb UGMhj"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.14 12.936C19.176 12.636 19.2 12.324 19.2 12C19.2 11.676 19.176 11.364 19.128 11.064L21.156 9.48002C21.336 9.33602 21.384 9.07202 21.276 8.86802L19.356 5.54402C19.236 5.32802 18.984 5.25602 18.768 5.32802L16.38 6.28802C15.876 5.90402 15.348 5.59202 14.76 5.35202L14.4 2.80802C14.364 2.56802 14.16 2.40002 13.92 2.40002H10.08C9.83998 2.40002 9.64799 2.56802 9.61199 2.80802L9.25199 5.35202C8.66398 5.59202 8.12399 5.91602 7.63199 6.28802L5.24398 5.32802C5.02798 5.24402 4.77598 5.32802 4.65598 5.54402L2.73598 8.86802C2.61598 9.08402 2.66398 9.33602 2.85598 9.48002L4.88398 11.064C4.83598 11.364 4.79998 11.688 4.79998 12C4.79998 12.312 4.82398 12.636 4.87198 12.936L2.84398 14.52C2.66398 14.664 2.61598 14.928 2.72398 15.132L4.64398 18.456C4.76398 18.672 5.01598 18.744 5.23198 18.672L7.61998 17.712C8.12398 18.096 8.65198 18.408 9.23998 18.648L9.59999 21.192C9.64798 21.432 9.83998 21.6 10.08 21.6H13.92C14.16 21.6 14.364 21.432 14.388 21.192L14.748 18.648C15.336 18.408 15.876 18.084 16.368 17.712L18.756 18.672C18.972 18.756 19.224 18.672 19.344 18.456L21.264 15.132C21.384 14.916 21.336 14.664 21.144 14.52L19.14 12.936ZM12 15.6C10.02 15.6 8.39998 13.98 8.39998 12C8.39998 10.02 10.02 8.40002 12 8.40002C13.98 8.40002 15.6 10.02 15.6 12C15.6 13.98 13.98 15.6 12 15.6Z"
                fill="#AEB3C8"
              />
            </svg>
            account
          </button>
        </li>
        <li>
          <button
            class="sc-bxivhb UGMhj"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 1.01L8 1C6.9 1 6 1.9 6 3V6H8V5H18V19H8V18H6V21C6 22.1 6.9 23 8 23H18C19.1 23 20 22.1 20 21V3C20 1.9 19.1 1.01 18 1.01ZM10 15H12V8H5V10H8.59L3 15.59L4.41 17L10 11.41V15Z"
                fill="#AEB3C8"
              />
            </svg>
            logout
          </button>
        </li>
      </ul>
    </nav>
    <main>
      <a
        href="/"
      >
        <img
          alt="Room Logo"
          src="logoText.svg"
        />
      </a>
      <div
        class="room-page"
      >
        <header
          class="room-page__header"
        >
          Chat
        </header>
        <div
          class="sc-bdVaJa bpitov"
        >
          <ul
            class="sc-htpNat AOzsL"
          >
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Kevin's avatar"
                class="user__avatar"
                src="https://source.unsplash.com/cqraK2a3Or8/50x50"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Kevin
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Kaitlyn's avatar"
                class="user__avatar"
                src="https://source.unsplash.com/tzzpfLiRPlA/50x50"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Kaitlyn
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Donte's avatar"
                class="user__avatar"
                src="https://source.unsplash.com/7nL3AhoUxw0/50x50"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Donte
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Chelsie's avatar"
                class="user__avatar"
                src="https://source.unsplash.com/4TMBk-LqEKo/50x50"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Chelsie
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Eriberto50's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/mactopus/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Eriberto50
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Leda7's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/ryanmclaughlin/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Leda7
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Davonte_Littel's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/wikiziner/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Davonte_Littel
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Dustin_Mante73's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/meelford/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Dustin_Mante73
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Ericka.Schmidt's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/oktayelipek/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Ericka.Schmidt
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Corbin82's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/dvdwinden/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Corbin82
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Pablo.DuBuque's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/dicesales/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Pablo.DuBuque
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Maye.Monahan36's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/ankitind/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Maye.Monahan36
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Modesta78's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/idiot/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Modesta78
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Holden.Roberts's avatar"
                class="user__avatar"
                src="https://s3.amazonaws.com/uifaces/faces/twitter/algunsanabria/128.jpg"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Holden.Roberts
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
            <li
              class="sc-bwzfXH edwCsR"
            >
              <img
                alt="Please just make one's avatar"
                class="user__avatar"
                src="https://via.placeholder.com/50x50"
              />
              <div
                class="user-glance"
              >
                <span
                  class="user-glance__name"
                >
                  Please just make one
                </span>
                <span
                  class="user-glance__last-message"
                >
                  Sure 8:pm
                </span>
              </div>
              <div
                class="message-info"
              >
                <span
                  class="message-info__last-contact"
                >
                  02 Feb
                </span>
                <span
                  class="message-info__notification"
                >
                  1
                </span>
              </div>
            </li>
          </ul>
          <div
            class="chatbox"
          >
            <h1
              class="chatbox__header"
            >
              test-room
            </h1>
            <ul
              class="chatbox__message-list"
            >
              <li
                class="chatbox__message-list--item"
              >
                ea quidem facilis
              </li>
              <li
                class="chatbox__message-list--item"
              >
                quaerat dolorum rerum
              </li>
              <li
                class="chatbox__message-list--item"
              >
                lorem ipsum dolor
              </li>
            </ul>
            <div
              class="chatbox__input"
            >
              <input
                class="chatbox__input--field"
                value=""
              />
              <button
                class="chatbox__input--button sc-bxivhb esJnqo"
                color="primary"
              >
                send
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Room renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-ifAKCX kViNAh"
  >
    <nav
      class="sc-EHOje hpFLPR"
    >
      <div
        class="user"
      >
        loading
      </div>
      <ul>
        <li>
          <button
            class="sc-bxivhb UGMhj"
            data-testid="create-room-button"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 20V14H14V20H19V12H22L12 3L2 12H5V20H10Z"
                fill="#AEB3C8"
              />
            </svg>
            Loading...
          </button>
        </li>
        <li>
          <button
            class="sc-bxivhb UGMhj"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.14 12.936C19.176 12.636 19.2 12.324 19.2 12C19.2 11.676 19.176 11.364 19.128 11.064L21.156 9.48002C21.336 9.33602 21.384 9.07202 21.276 8.86802L19.356 5.54402C19.236 5.32802 18.984 5.25602 18.768 5.32802L16.38 6.28802C15.876 5.90402 15.348 5.59202 14.76 5.35202L14.4 2.80802C14.364 2.56802 14.16 2.40002 13.92 2.40002H10.08C9.83998 2.40002 9.64799 2.56802 9.61199 2.80802L9.25199 5.35202C8.66398 5.59202 8.12399 5.91602 7.63199 6.28802L5.24398 5.32802C5.02798 5.24402 4.77598 5.32802 4.65598 5.54402L2.73598 8.86802C2.61598 9.08402 2.66398 9.33602 2.85598 9.48002L4.88398 11.064C4.83598 11.364 4.79998 11.688 4.79998 12C4.79998 12.312 4.82398 12.636 4.87198 12.936L2.84398 14.52C2.66398 14.664 2.61598 14.928 2.72398 15.132L4.64398 18.456C4.76398 18.672 5.01598 18.744 5.23198 18.672L7.61998 17.712C8.12398 18.096 8.65198 18.408 9.23998 18.648L9.59999 21.192C9.64798 21.432 9.83998 21.6 10.08 21.6H13.92C14.16 21.6 14.364 21.432 14.388 21.192L14.748 18.648C15.336 18.408 15.876 18.084 16.368 17.712L18.756 18.672C18.972 18.756 19.224 18.672 19.344 18.456L21.264 15.132C21.384 14.916 21.336 14.664 21.144 14.52L19.14 12.936ZM12 15.6C10.02 15.6 8.39998 13.98 8.39998 12C8.39998 10.02 10.02 8.40002 12 8.40002C13.98 8.40002 15.6 10.02 15.6 12C15.6 13.98 13.98 15.6 12 15.6Z"
                fill="#AEB3C8"
              />
            </svg>
            account
          </button>
        </li>
        <li>
          <button
            class="sc-bxivhb UGMhj"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 1.01L8 1C6.9 1 6 1.9 6 3V6H8V5H18V19H8V18H6V21C6 22.1 6.9 23 8 23H18C19.1 23 20 22.1 20 21V3C20 1.9 19.1 1.01 18 1.01ZM10 15H12V8H5V10H8.59L3 15.59L4.41 17L10 11.41V15Z"
                fill="#AEB3C8"
              />
            </svg>
            logout
          </button>
        </li>
      </ul>
    </nav>
    <main>
      <a
        href="/"
      >
        <img
          alt="Room Logo"
          src="logoText.svg"
        />
      </a>
      <div
        class="room-page"
      >
        <header
          class="room-page__header"
        >
          Chat
        </header>
        loading...
      </div>
    </main>
  </div>
</DocumentFragment>
`;
